examples/07_advanced/plot_mask_large_fmri.py

```diff
@@ -116,61 +110,73 @@
         affine=resampled_atlas.affine,
         copy_header=True,
     )
-
-    mask_imgs.append(mask)
     path = output_dir / f"mask_{idx}.nii.gz"
     mask.to_filename(path)
     mask_paths.append(path)
 
+# remove images from memory
+del atlas_img, resampled_atlas, mask, fmri_img
+
 # %%
 # Mask the fMRI image using NiftiMasker
 # -------------------------------------
-# Let's first see how we would typically mask the fMRI image using
-# :class:`~nilearn.maskers.NiftiMasker`. This is the most common way to extract
-# data from an fMRI image as it makes it easy to standardize, smooth, detrend,
-# etc. the data.
+# Let's first mask the fMRI image using :class:`~nilearn.maskers.NiftiMasker`.
+# This is the most user-friendly way to extract data from an fMRI image as it
+# makes it easy to standardize, smooth, detrend, etc. the data.
 #
 # We will first wrap the :func:`nilearn.maskers.NiftiMasker.fit_transform`
 # within a function so that it is more readable and easier to use.
 # We will then define another function that would mask the fMRI image using
-# multiple masks in parallel using the :mod:`joblib` package.
+# multiple masks in parallel using the :mod:`joblib` package. As mentioned
+# earlier, this could further be done in two ways: directly using file paths
+# as input or first loading the images in-memory and passing them as input.
+#
+# So we will define two functions for each case.
 #
-# We can then use this function to measure the memory usage using the
+# We can then track the memory usage of these functions via the
 # ``memory_profiler`` package.
 
 from joblib import Parallel, delayed
 
 from nilearn.maskers import NiftiMasker
 
 
-def nifti_masker_single(fmri_path, mask_path):
-    return NiftiMasker(mask_img=mask_path).fit_transform(fmri_path)
+def nifti_masker_single(fmri, mask):
+    return NiftiMasker(mask_img=mask).fit_transform(fmri)
+
+
+def nifti_masker_parallel_path(fmri_path, mask_paths):
+    Parallel(n_jobs=N_REGIONS)(
+        delayed(nifti_masker_single)(fmri_path, mask_path)
+        for mask_path in mask_paths
+    )
+    return 1
```

Comment:
Just to make sure I understand : Why are we adding these `return` statements ?

Reply:
Nothing specific, I just wanted to avoid returning the output of Parallel and later storing it in a variable because that would show up as an irrelavant memory peak.